src/services/routeRegex.spec.ts

```diff
@@ -50,6 +53,23 @@ describe('generateRoutePathRegexPattern', () => {
     const expected = new RegExp(`^parent/child/${catchAll}/grand-child/${catchAll}$`, 'i')
     expect(result.toString()).toBe(expected.toString())
   })
+
+
+  test('given path with default params, returns value with params replaced with catchall', () => {
```

Comment:
this should work exactly the same as any other param right? Do we need to test this? The regex doesn't factor in the param right?

Reply:
correct, this was from my first pass at defaults where it also made them optional. I will remove. Good catch